### PR TITLE
Miscellaneous infrastructure and fixes for FAR

### PR DESCRIPTION
**Pull request by rynowak**
Working through the FAR changes and biting off small pieces that can be merged independently 👍 

See commits for details.

### CODE REVIEW

File: src/Microsoft.CodeAnalysis.Razor.Workspaces/RazorDocumentExcerptService.cs [hunk 1]
```diff
@@ -89,24 +100,26 @@ public RazorDocumentExcerptService(DocumentSnapshot document, ISpanMappingServic
...
-
-        private TextSpan ChooseExcerptSpan(SourceText primaryText, TextSpan primarySpan, ExcerptMode mode)
+        
+        private TextSpan ChooseExcerptSpan(SourceText primaryText, TextSpan primarySpan, ExcerptModeInternal mode)
         {
             var startLine = primaryText.Lines.GetLineFromPosition(primarySpan.Start);
             var endLine = primaryText.Lines.GetLineFromPosition(primarySpan.End);
 
-            // If we're showing a single line then this will do. Otherwise expand the range by 1 in
+            // If we're showing a single line then this will do. Otherwise expand the range by 3 in
```
Comment (ajaybhargavb) [inline src/Microsoft.CodeAnalysis.Razor.Workspaces/RazorDocumentExcerptService.cs]: Why the change from 1 to 3? Is that what Roslyn does in other places?

Comment (rynowak) [reply src/Microsoft.CodeAnalysis.Razor.Workspaces/RazorDocumentExcerptService.cs]: Yeah, HeeJae recommend 3.

File: src/Microsoft.VisualStudio.LanguageServices.Razor/ProjectSystem/GeneratedDocumentTextLoader.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        public GeneratedDocumentTextLoader(DocumentSnapshot document, string filePath)
+        {
+            if (document == null)
+            {
+                throw new ArgumentNullException(nameof(document));
+            }
+
+            _document = document;
+            _filePath = filePath;
+            _version = VersionStamp.Create();
```
Comment (NTaylorMullen) [inline src/Microsoft.VisualStudio.LanguageServices.Razor/ProjectSystem/GeneratedDocumentTextLoader.cs]: Shouldn't this be looking at the documents version and creating a version that's newer?

Comment (rynowak) [reply src/Microsoft.VisualStudio.LanguageServices.Razor/ProjectSystem/GeneratedDocumentTextLoader.cs]: No, that version isn't meaningful to Roslyn. They also don't use `VersionStamp` to indicate *newer*, they only use `VersionStamp` to indicate *different*
